homeassistant/components/motion_blinds/config_flow.py

```diff
@@ -29,35 +35,67 @@ class MotionBlindsFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
 
     def __init__(self):
         """Initialize the Motion Blinds flow."""
-        self.host = None
-        self.key = None
+        self._host = None
+        self._key = None
+        self._ips = []
 
     async def async_step_user(self, user_input=None):
         """Handle a flow initialized by the user."""
         errors = {}
         if user_input is not None:
-            self.host = user_input[CONF_HOST]
-            self.key = user_input[CONF_API_KEY]
-            return await self.async_step_connect()
+            self._host = user_input.get(CONF_HOST)
+
+            if self._host is not None:
+                return await self.async_step_connect()
+
+            # Use MotionGateway discovery
+            discover_class = MotionDiscovery()
+            gateways = await self.hass.async_add_executor_job(discover_class.discover)
+            self._ips = list(gateways)
+
+            if len(self._ips) == 1:
+                self._host = self._ips[0]
+                return await self.async_step_connect()
+
+            if len(self._ips) > 1:
+                return await self.async_step_select()
+
+            errors["base"] = "discovery_error"
 
         return self.async_show_form(
             step_id="user", data_schema=CONFIG_SCHEMA, errors=errors
         )
 
+    async def async_step_select(self, user_input=None):
+        """Handle multiple motion gateways found."""
+        if user_input is not None:
+            self._host = user_input["select_ip"]
+            return await self.async_step_connect()
+
+        select_schema = vol.Schema({vol.Required("select_ip"): vol.In(self._ips)})
+
+        return self.async_show_form(step_id="select", data_schema=select_schema)
+
     async def async_step_connect(self, user_input=None):
         """Connect to the Motion Gateway."""
+        if user_input is not None:
+            self._key = user_input[CONF_API_KEY]
```

Comment:
I think _key can be a local variable, rather than an instance level variable since it is only used within this step.

Reply:
Happy to get this merged today after you take a look at this one.

Reply:
thanks, I made the change.